src/platform/uefi_aarch64/main.zig

```diff
@@ -0,0 +1,369 @@
+pub const sabaton = @import("../../sabaton.zig");
+
+const std = @import("std");
+const uefi = std.os.uefi;
+
+const fs = @import("fs.zig");
+
+var conout: ?*uefi.protocols.SimpleTextOutputProtocol = null;
+
+pub const panic = sabaton.panic;
+
+pub const io = struct {
+    pub fn putchar(ch: u8) void {
+        if (conout) |co| {
+            if (ch == '\n')
+                putchar('\r');
+            const chrarr = [2]u16{ ch, 0 };
+            _ = co.outputString(@ptrCast(*const [1:0]u16, &chrarr));
+        }
+    }
+};
+
+pub fn locateProtocol(comptime T: type) callconv(.Inline) ?*T {
+    var ptr: *T = undefined;
+    const guid: std.os.uefi.Guid align(8) = T.guid;
+    if (.Success != uefi.system_table.boot_services.?.locateProtocol(&guid, null, @ptrCast(*?*c_void, &ptr))) {
+        return null;
+    }
+    return ptr;
+}
+
+pub fn handleProtocol(handle: uefi.Handle, comptime T: type) callconv(.Inline) ?*T {
+    var ptr: *T = undefined;
+    const guid: std.os.uefi.Guid align(8) = T.guid;
+    if (.Success != uefi.system_table.boot_services.?.handleProtocol(handle, &guid, @ptrCast(*?*c_void, &ptr))) {
```

Comment:
Why is the order inverted here?

Reply:
Because where I stole it from had it in that order :^)